src/convertToObject.js

```diff
@@ -6,7 +6,28 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.trim()) {
+    return {};
+  }
+
+  const result = {};
+
+  const rules = sourceString
+    .split(';')
+    .map((rule) => rule.trim())
+    .filter(Boolean);
+
+  rules.forEach((rule) => {
+    const [key, ...valueParts] = rule.split(':');
+
+    if (key && valueParts.length > 0) {
+      const value = valueParts.join(':').trim();
+
+      result[key.trim()] = value;
+    }
```

Comment:
The condition `if (key && valueParts.length > 0)` checks if both key and value parts exist, which is crucial for avoiding errors when processing malformed input. This is a good validation step.